src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -114,5 +117,30 @@ public void addMessage(List<Message> messages, Entity entity, String user) {
 
   return messages;
  }
+  /** 
+   * Gets all the users
+   * 
+   * @return a set of users
+   */
+  public Set<String> getUsers(){
+    Set<String> users = new HashSet<>();
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    for(Entity entity : results.asIterable()) {
+      users.add((String) entity.getProperty("user"));
+    }
+    return users;
+  }
 
+  /** Returns the total number of messages for all users. */
+  public int getTotalMessageCount(){
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    return results.countEntities(FetchOptions.Builder.withLimit(1000));
+  }
+
+  /** Returns the total number of users. */
+  public int getTotalUserCount(){
+    return getUsers().size();
```

Comment:
FYI (no action required) : Getting the all the users and looking at the size should be ok for now. But also think about performance if the number of users are large. Later on you could look into whether datastore provides any queries you could use to get the count more efficiently.